lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -1220,7 +1220,12 @@ PrimarySkill::PrimarySkill CGHeroInstance::nextPrimarySkill(CRandomGenerator & r
 			break;
 		}
 	}
-
+	if(primarySkill >= GameConstants::PRIMARY_SKILLS)
+	{
```

Comment:
I would suggest to either retry the logic above or normalize probabilities right after they are loaded from json

Reply:
Or automatically select the skill with highest probability

Reply:
Also we can sum up all probabilities and use this value instead of 99

Reply:
I've added some error messages to the mod loading routine.

In general case,, it may be zero or negative value. Thus, I don't want to think instead of mods authors at this point and invent some kind of crutches. My goal here is to warn mod developer and provide appropriate game experience for player.

Reply:
Agreed.